src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Browser.cs

```diff
@@ -9,6 +10,10 @@ namespace System.Net.WebSockets
     internal sealed class WebSocketHandle
     {
         private WebSocketState _state = WebSocketState.Connecting;
+        public HttpStatusCode HttpStatusCode { get; set; }
+
+        // setter to clean up when not needed anymore
+        public IReadOnlyDictionary<string, IEnumerable<string>>? HttpResponseHeaders { get; set; }
```

Comment:
It looks like if someone sets CollectHttpResponseDetails to true when using the browser implementation, they won't get any errors but HttpStatusCode won't be properly set and this headers collection will remain null.  Yes?  We should probably make the browser implementation throw PNSE if CollectHttpResponseDetails is set to true.

Reply:
Yes, it is true, because I was following approved API which doesn't mention unsupported browser. Sound reasonable to me, I would throw exception for headers. What for `HttpStatusCode` - I don't see it in WebSocketHandle.Browser, did you mean `closeStatus`?

Reply:
> I don't see it in WebSocketHandle.Browser, did you mean closeStatus?

I mean the "public HttpStatusCode HttpStatusCode { get; set; }" you're adding here.

Reply:
I believe no HttpResponse details are available on browser. It is all hidden inside js library. (@pavelsavara might be able to share more insights). So I believe the best approach is to make both status code and headers to be under the flag (which you can't set on browser).

> if someone sets CollectHttpResponseDetails to true

I am not sure it is possible? CollectHttpResponseDetails setter should throw PNSE on browser, if I don't miss something
https://github.com/dotnet/runtime/blob/adea0cc8f74cc80f6379e21253aae170aa65d5c1/src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/ClientWebSocketOptions.cs#L70-L74

Reply:
Confirmed with Pavel that we cannot get status code on browser. I would prefer to leave status code under the flag in that case. The initial comment was before I set PNSE for `CollectHttpResponseDetails`.